### PR TITLE
storage: separate samples into standalone files (generated by script)

### PR DESCRIPTION
**Pull request by IlyaFaer**
Made by script. Every sample moved into standalone file. Names of files writen by samples names (as docs require). Imports are made with `goimports -w .`. Start and end tags moved to starts and ends of files

`buckets` directory

### CODE REVIEW

**Comment by IlyaFaer** [comment]
@AlisskaPie

File: storage/buckets/checkRequesterPays.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	attrs, err := client.Bucket(bucketName).Attrs(ctx)
+	if err != nil {
+		return err
+	}
+	log.Printf("Is requester pays enabled? %v\n", attrs.RequesterPays)
```
Comment (IlyaFaer) [inline storage/buckets/checkRequesterPays.go]: I'm not very sure, as in here we're printing value which we got, but this should probably be done through writer

image
image

File: storage/buckets/create.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+package buckets
+
+// [START create_bucket]
+import (
+	"context"
+
+	"cloud.google.com/go/storage"
+)
+
+// create creates the bucket in the project.
```
Comment (IlyaFaer) [inline storage/buckets/create.go]: Better `creates new bucket`

File: storage/buckets/createWithAttrs.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+// [START create_bucket_with_storageclass_and_location]
+import (
+	"context"
+
+	"cloud.google.com/go/storage"
+)
+
+// createWithAttrs creates the bucket in the project with storage class and
+// location.
```
Comment (IlyaFaer) [inline storage/buckets/createWithAttrs.go]: `Storage` with upper first letter. Create **new** bucket

File: storage/buckets/enableRequesterPays.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+package buckets
+
+// [START enable_requester_pays]
+import (
+	"context"
+
+	"cloud.google.com/go/storage"
+)
+
+// enableRequesterPays sets requester pays status to true.
```
Comment (IlyaFaer) [inline storage/buckets/enableRequesterPays.go]: Better say "flag" than "status"

File: storage/buckets/getDefaultEventBasedHold.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+// [START storage_get_default_event_based_hold]
+import (
+	"context"
+	"log"
+
+	"cloud.google.com/go/storage"
+)
+
+// getDefaultEventBasedHold gets default event-based hold.
```
Comment (IlyaFaer) [inline storage/buckets/getDefaultEventBasedHold.go]: Let's add `state` to the end of the sentence

File: storage/buckets/get_bucket_metadata.go [hunk 6]
```diff
@@ -16,7 +16,7 @@
 // using the Google Storage API. More documentation is available at
 // https://cloud.google.com/storage/docs/json_api/v1/.
 
-package main
+package buckets
```
Comment (IlyaFaer) [inline storage/buckets/get_bucket_metadata.go:19]: Don't need an empty line before `package`

File: storage/buckets/removeRetentionPolicy.go [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+func removeRetentionPolicy(bucketName string) error {
+	// bucketName := "bucket-name"
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	bucket := client.Bucket(bucketName)
+
```
Comment (IlyaFaer) [inline storage/buckets/removeRetentionPolicy.go]: Let's delete whitelines

File: storage/buckets/setDefaultKMSkey.go [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+package buckets
+
+// [START storage_set_bucket_default_kms_key]
+import (
+	"context"
+
+	"cloud.google.com/go/storage"
+)
+
+// setDefaultKMSkey sets the Cloud KMS encryption key.
```
Comment (IlyaFaer) [inline storage/buckets/setDefaultKMSkey.go]: Let's add `for bucket`

File: storage/buckets/setDefaultKMSkey.go [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+func setDefaultKMSkey(bucketName string, keyName string) error {
```
Comment (IlyaFaer) [inline storage/buckets/setDefaultKMSkey.go]: Type can be collapsed into one word

**Review by IlyaFaer — COMMENTED** [review]
And we should change `2019` to `2020` in license lines

File: storage/buckets/create.go (hunk 2 above)
Comment (AlisskaPie) [reply storage/buckets/create.go]: Done

File: storage/buckets/createWithAttrs.go (hunk 3 above)
Comment (AlisskaPie) [reply storage/buckets/createWithAttrs.go]: Done

File: storage/buckets/enableRequesterPays.go (hunk 4 above)
Comment (AlisskaPie) [reply storage/buckets/enableRequesterPays.go]: Done

File: storage/buckets/getDefaultEventBasedHold.go (hunk 5 above)
Comment (AlisskaPie) [reply storage/buckets/getDefaultEventBasedHold.go]: Done

File: storage/buckets/get_bucket_metadata.go (hunk 6 above)
Comment (AlisskaPie) [reply storage/buckets/get_bucket_metadata.go:19]: Done

File: storage/buckets/removeRetentionPolicy.go (hunk 7 above)
Comment (AlisskaPie) [reply storage/buckets/removeRetentionPolicy.go]: Done

File: storage/buckets/setDefaultKMSkey.go (hunk 8 above)
Comment (AlisskaPie) [reply storage/buckets/setDefaultKMSkey.go]: Done

File: storage/buckets/setDefaultKMSkey.go (hunk 9 above)
Comment (AlisskaPie) [reply storage/buckets/setDefaultKMSkey.go]: Done

File: storage/buckets/buckets_test.go [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+	tc := testutil.SystemTest(t)
+	setup(t)
+
+	buckets, err := list(tc.ProjectID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var ok bool
+outer:
```
Comment (IlyaFaer) [inline storage/buckets/buckets_test.go]: What's this?

File: storage/buckets/getDefaultEventBasedHold.go [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+		return nil, err
+	}
+	defer client.Close()
+
+	attrs, err := client.Bucket(bucketName).Attrs(ctx)
+	if err != nil {
+		return nil, err
+	}
+	fmt.Fprintf(w, "Default event-based hold enabled? %t\n",
+		attrs.DefaultEventBasedHold)
```
Comment (IlyaFaer) [inline storage/buckets/getDefaultEventBasedHold.go]: Was it done by `gofmt`? Looks a little bit strange

File: storage/buckets/getUniformBucketLevelAccess.go [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+	attrs, err := client.Bucket(bucketName).Attrs(ctx)
+	if err != nil {
+		return nil, err
+	}
+	uniformBucketLevelAccess := attrs.UniformBucketLevelAccess
+	if uniformBucketLevelAccess.Enabled {
+		fmt.Fprintf(w, "Uniform bucket-level access is enabled for %q.\n",
+			attrs.Name)
+		fmt.Fprintf(w, "Bucket will be locked on %q.\n",
+			uniformBucketLevelAccess.LockedTime)
```
Comment (IlyaFaer) [inline storage/buckets/getUniformBucketLevelAccess.go]: This also looks strange? `gofmt`? I think, we should try to write it manually  like:
```
Fprintf(
    w,
    "Uniform ...",
    attrs.Name
)
```
And then run `gofmt` over it

File: storage/buckets/get_bucket_metadata.go [hunk 13]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2019 Google LLC
+// Copyright 2020 Google LLC
```
Comment (IlyaFaer) [inline storage/buckets/get_bucket_metadata.go]: This file name isn't camelCased

File: storage/buckets/lockRetentionPolicy.go [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+	}
+
+	lockedAttrs, err := client.Bucket(bucketName).Attrs(ctx)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(w, "Retention policy for %v is now locked\n", bucketName)
+	fmt.Fprintf(w, "Retention policy effective as of %v\n",
+		lockedAttrs.RetentionPolicy.EffectiveTime)
```
Comment (IlyaFaer) [inline storage/buckets/lockRetentionPolicy.go]: Strange paddings too

**Review by IlyaFaer — COMMENTED** [review]
We also should delete `main.go` file and probably `main_test.go` as tests are now living in `objects_test.go`

File: storage/buckets/lockRetentionPolicy.go (hunk 14 above)
Comment (AlisskaPie) [reply storage/buckets/lockRetentionPolicy.go]: I used `gofmt` for every file before loading
I agree, it looks strange

File: storage/buckets/getDefaultEventBasedHold.go (hunk 11 above)
Comment (AlisskaPie) [reply storage/buckets/getDefaultEventBasedHold.go]: Done

File: storage/buckets/getUniformBucketLevelAccess.go (hunk 12 above)
Comment (AlisskaPie) [reply storage/buckets/getUniformBucketLevelAccess.go]: Done

File: storage/buckets/get_bucket_metadata.go (hunk 13 above)
Comment (AlisskaPie) [reply storage/buckets/get_bucket_metadata.go]: Done

File: storage/buckets/addBucketIamMember.go [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	bucket := client.Bucket(bucketName)
+	policy, err := bucket.IAM().Policy(ctx)
+	if err != nil {
+		return fmt.Errorf("Bucket(%q).IAM.Policy: %v", bucketName, err)
```
Comment (IlyaFaer) [inline storage/buckets/addBucketIamMember.go]: I think, we should write `Bucket(%q).IAM().Policy` as IAM is method too

File: storage/buckets/buckets_test.go [hunk 16]
```diff
@@ -0,0 +1,256 @@
...
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+func TestMain(m *testing.M) {
+	s := m.Run()
+	os.Exit(s)
+}
```
Comment (IlyaFaer) [inline storage/buckets/buckets_test.go]: We don't need this function, as we're not doing anything more than running tests in it

File: storage/buckets/buckets_test.go [hunk 17]
```diff
@@ -0,0 +1,256 @@
...
+	if err := createBucket(tc.ProjectID, bucketName); err != nil {
+		t.Fatalf("createBucket: %v", err)
+	}
+}
+
+func TestCreateBucketClassLocation(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	name := tc.ProjectID + "-storage-buckets-tests-attrs"
+
+	// Clean up bucket before running test.
```
Comment (IlyaFaer) [inline storage/buckets/buckets_test.go]: `the test` I believe

File: storage/buckets/buckets_test.go [hunk 18]
```diff
@@ -0,0 +1,256 @@
...
+	}
+}
+
+func TestListBuckets(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	bucketName := tc.ProjectID + "-storage-buckets-tests"
+
+	buckets, err := listBuckets(tc.ProjectID)
+	if err != nil {
+		t.Fatal(err)
```
Comment (IlyaFaer) [inline storage/buckets/buckets_test.go]: No last method name in `Fatal()`. It's the place, where method name will be duplicated or something?

File: storage/buckets/buckets_test.go [hunk 19]
```diff
@@ -0,0 +1,256 @@
...
+	bucketName := tc.ProjectID + "-storage-buckets-tests"
+
+	buf := new(bytes.Buffer)
+	if _, err := getBucketMetadata(buf, bucketName); err != nil {
+		t.Errorf("getBucketMetadata: %#v", err)
+	}
+
+	got := buf.String()
+	if want := "BucketName:"; !strings.Contains(got, want) {
+		t.Errorf("contents = %q, want %q", got, want)
```
Comment (IlyaFaer) [inline storage/buckets/buckets_test.go]: I'd change `contents` to `got`, though I don't think it's critical

File: storage/buckets/removeBucketIamMember.go [hunk 20]
```diff
@@ -0,0 +1,53 @@
...
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	bucket := client.Bucket(bucketName)
+	policy, err := bucket.IAM().Policy(ctx)
+	if err != nil {
+		return fmt.Errorf("Bucket(%q).IAM.Policy: %v", bucketName, err)
```
Comment (IlyaFaer) [inline storage/buckets/removeBucketIamMember.go]: `IAM()` too

File: storage/buckets/main_test.go [hunk 21]
```diff
@@ -1,270 +0,0 @@
...
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package main
-
-import (
```
Comment (IlyaFaer) [inline storage/buckets/main_test.go:17]: I suppose, you just renamed this file!? We're not deleting old files for now

**Review by IlyaFaer — COMMENTED** [review]
Looking good, just a couple of nits, and we're ready to push it publically

File: storage/buckets/addBucketIamMember.go (hunk 15 above)
Comment (AlisskaPie) [reply storage/buckets/addBucketIamMember.go]: Done

File: storage/buckets/buckets_test.go (hunk 16 above)
Comment (AlisskaPie) [reply storage/buckets/buckets_test.go]: Done

File: storage/buckets/buckets_test.go (hunk 17 above)
Comment (AlisskaPie) [reply storage/buckets/buckets_test.go]: Ok, let it be

File: storage/buckets/buckets_test.go (hunk 18 above)
Comment (AlisskaPie) [reply storage/buckets/buckets_test.go]: Done

File: storage/buckets/buckets_test.go (hunk 19 above)
Comment (AlisskaPie) [reply storage/buckets/buckets_test.go]: Done

File: storage/buckets/removeBucketIamMember.go (hunk 20 above)
Comment (AlisskaPie) [reply storage/buckets/removeBucketIamMember.go]: Done

File: storage/buckets/main_test.go (hunk 21 above)
Comment (AlisskaPie) [reply storage/buckets/main_test.go:17]: Right, this file now has name buckets_test.go

**Pull request closed without merging**